### PR TITLE
Fix suggested "gpg" usage to stop relying on deprecated and insecure behavior

### PR DESCRIPTION
**Pull request by tianon**
cc @aburch -- does this look at least _more_ correct? :innocent: (anything we could do to make this pattern even better?)

### CODE REVIEW

**Comment by tianon** [comment]
(Wanna fix it properly here at the source before replicating outwards.)

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ RUN set -x \
 	&& curl -fSL "https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini.asc" -o /usr/local/bin/tini.asc \
 	&& export GNUPGHOME="$(mktemp -d)" \
```
Comment (md5) [inline README.md]: If this `GNUPGHOME` export is a best practice, it seems like it should be in the `python.tar.xz.asc` example below as well.

Comment (tianon) [inline README.md]: Yeah, that's a good point -- I was going to update that separately, but might as well include it here too.

Comment (tianon) [inline README.md]: Updated. :smile:

Comment (md5) [inline README.md]: :+1:

**Comment by yosifkit** [comment]
LGTM

**Comment by tianon** [comment]
I just realized we probably ought to link to the source of _where_ this is pointed out to be "insecure"...  :innocent:

From `gpg(1)` (http://manpages.debian.org/cgi-bin/man.cgi?query=gpg&manpath=Debian+unstable+sid):

>        --verify
>
>     ...
>
>               Note: If the option --batch is not used, gpg may assume  that  a
>               single  argument is a file with a detached signature and it will
>               try to find a matching data file by stripping certain  suffixes.
>               Using  this historical feature to verify a detached signature is
>               strongly discouraged; always specify the data file too.

**Pull request merged**